File: atlas-extract-server/Dtos/ExtractionStartReq.cs
namespace App.Dtos;



public class ExtractionStartReq
{
    public required string SourceId { get; set; }
    public required PageRange[] PageRanges { get; set; }
    public required string SourceLanguage { get; set; }
    public required string SourceTopic { get; set; }
    public required string StructureDescription { get; set; }
    public string Ignore { get; set; } = "";
    public required string DescriptionLength { get; set; }
    public string AdditionalInstructions { get; set; } = "";
}

File: atlas-extract-client/src/app/types/ExtractionStartReq.ts
export interface ExtractStartReq {
    pageRanges: {startPage: number; endPage: number}[];
    friendlyName: string;
    sourceId: string;
    sourceLanguage: string;
    sourceTopic: string;
    structureDescription: string;
    ignore?: string;
    descriptionLength: string;
    additionalInstructions?: string;
}